core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -829,12 +873,28 @@ private[spark] class ExecutorAllocationManager(
       numTotalTasks - numRunning
     }
 
+    // Currently TaskSetManager.getCompletelyBlacklistedTaskIfAny only takes the first unschedulable
+    // task found due to blacklisting. This way we only need to keep track of the unschedulable
+    // tasksets which is an indirect way to get the current number of unschedulable tasks.
+    def pendingUnschedulableTaskSetsPerResourceProfile(rp: Int): Int = {
+      val attempts = resourceProfileIdToStageAttempt.getOrElse(rp, Set.empty).toSeq
+      attempts.filter(attempt => unschedulableTaskSets.contains(attempt)).size
+    }
+
+    def hasPendingUnschedulableTasks: Boolean = {
+      val attemptSets = resourceProfileIdToStageAttempt.values
+      attemptSets.exists { attempts =>
+        attempts.exists(unschedulableTaskSets.contains(_))
+      }
+    }
+
     def hasPendingTasks: Boolean = {
-      hasPendingSpeculativeTasks || hasPendingRegularTasks
+      hasPendingSpeculativeTasks || hasPendingRegularTasks || hasPendingUnschedulableTasks
```

Comment:
this doesn't make sense to me. Did you have a specific bug without this?  
The tasks themselves are counted as regular tasks.

Reply:
No, this is not needed. I will remove this. Thanks.